CI: create GitHub Release, multi-platform build, fix linux workflow

llerafel opened the pull request:
### Changes

#### `linux.yml`
- **Added** matrix strategy (amd64/arm64) 
- **Added** xmake dependency caching
- **Fix** artifact upload step with `upload-build-for-next-job`

#### `windows.yml`
- **Merge** prepare package from `playble-build`

#### `windows-playable-build.yml` → `playable-build.yml`
- **Added** `prepare` job for version extraction and conditions `Disable cron startup in forks`
- **Added** calling `linux.yml`
- **Addd** re-upload artifact (rename) in matrix
- **Added** `release-upload` job on tag pushes

---

### Now

```mermaid
---
config:
  layout: dagre
---
flowchart LR
    Prepare["Prepare job<br>(get version)"] --> CheckFork["Fork + schedule?"]
    CheckFork -- Yes --> Skip["Skip build"]
    CheckFork -- No --> Build["Build jobs<br>(Windows + Linux)"]
    ReleaseCheck["Is tag push?"] -- Yes --> Release["Create GitHub Release"]
    ReleaseCheck -- No --> n1["Just upload artifact"]
    Build --> ReleaseCheck
    n2["Start"] --> Prepare

    CheckFork@{ shape: rounded}
    ReleaseCheck@{ shape: terminal}
    n1@{ shape: rect}
    n2@{ shape: start}
     Prepare:::ProcessNode
     CheckFork:::DecisionNode
     Skip:::SkipNode
     Build:::WorkflowNode
     ReleaseCheck:::DecisionNode
     Release:::ProcessNode
     n1:::ProcessNode
    classDef StartNode stroke-width:2px,stroke:#4CAF50,fill:#E8F5E9,color:#2E7D32
    classDef EndNode stroke-width:2px,stroke:#F44336,fill:#FFEBEE,color:#C62828
    classDef DecisionNode stroke-width:2px,stroke:#FF9800,fill:#FFF3E0,color:#E65100
    classDef ProcessNode stroke-width:2px,stroke:#2196F3,fill:#E3F2FD,color:#1565C0
    classDef WorkflowNode stroke-width:2px,stroke:#9C27B0,fill:#F3E5F5,color:#6A1B9A
    classDef ArtifactNode stroke-width:2px,stroke:#00BCD4,fill:#E0F7FA,color:#00838F
    classDef SkipNode stroke-width:2px,stroke:#9E9E9E,fill:#FAFAFA,color:#424242
```

---

### Pre-merge
- [x] ~~[Test tag v9.9.9](https://github.com/llerafel/TiltedEvolution/releases/tag/v9.9.9)~~, [New test tag v9.9.8](https://github.com/llerafel/TiltedEvolution/releases/tag/v9.9.8)
- [x] Verify multi-platform builds (linux-arm64)
- [ ] **Rename** `windows-playable-build.yml` to `playable-build.yml`
- [ ] **Fix** `README` badge link
```markdown
![Build status](https://github.com/tiltedphoques/TiltedEvolution/actions/workflows/playable-build.yml/badge.svg?branch=master)
```

#### GHA test run
- [New tag](https://github.com/llerafel/TiltedEvolution/actions/runs/20190519098)
- [Manual=cron](https://github.com/llerafel/TiltedEvolution/actions/runs/20193723719)

llerafel committed:
Fix tabulation

llerafel committed:
Fix linux ci

llerafel committed:
migrate package prepare from playable-build

llerafel committed:
Adding release and re-upload artifact logic

llerafel committed:
fix job name

llerafel committed:
cleanup junk step

llerafel committed:
fix re-upload arifact step

llerafel committed:
fix

llerafel committed:
Fix re-upload logic

llerafel committed:
Fix release job

llerafel committed:
version in filename artifact

llerafel committed:
Fix re-upload zip step

llerafel committed:
filename 😕

llerafel committed:
directory 😐

miredirex commented on .github/workflows/linux.yml:
```diff
@@ -5,56 +5,89 @@ on:
   push:
     branches:
       - master
+      - prerel
```
'prerel' branch hasn't been used in years, you can remove this

miredirex commented on .github/workflows/windows-playable-build.yml:
```diff
@@ -12,43 +12,86 @@ on:
+          merge-multiple: true
+
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v1
```
Let's not use a third-party action when `gh release create` exists. `gh` is pre-installed in gha runners

miredirex commented on .github/workflows/linux.yml:
```diff
@@ -5,56 +5,89 @@ on:
+      - name: Checkout master on tag push
+        if: github.ref_type == 'tag'
+        # Checkout only if the tag was pushed to master
+        run: '[ "$(git rev-parse HEAD)" = "$(git rev-parse origin/master)" ] && git checkout master || echo False'
```
Why the `echo`?

miredirex commented on .github/workflows/linux.yml:
```diff
@@ -5,56 +5,89 @@ on:
+          actions-cache-key: ${{ matrix.runner }}
+
+      # Setup compilation mode and install project dependencies
+      # (continue-on-error + timeout is a temporary solution until sentry-native is fixed; shouldn't affect the building step)
```
Isn't this comment outdated now? You fixed the hanging issue, right?

miredirex commented on .github/workflows/linux.yml:
```diff
@@ -5,56 +5,89 @@ on:
+          mkdir -p ${{ env.STR_BUILD_DIR }}
+          cp packaged/lib/libSTServer.so ${{ env.STR_BUILD_DIR }}/ || true
+          cp packaged/bin/crashpad_handler ${{ env.STR_BUILD_DIR }}/ || true
+          cp packaged/bin/SkyrimTogetherServer ${{ env.STR_BUILD_DIR }}/ || true
```
`|| true`? I think a job should fail if the file wasn't found during this step

miredirex commented on .github/workflows/linux.yml:
```diff
@@ -5,56 +5,89 @@ on:
+          path: |
+            ${{ env.STR_BUILD_DIR }}/libSTServer.so
+            ${{ env.STR_BUILD_DIR }}/crashpad_handler
+            ${{ env.STR_BUILD_DIR }}/SkyrimTogetherServer
```
```suggestion
          path: ${{ env.STR_BUILD_DIR }}
```

llerafel committed:
Remove pre-release branch

llerafel committed:
Enforce master branch check for tag releases

llerafel committed:
Clean-up outdate comment

llerafel committed:
Remove outdate fallback

llerafel committed:
Use GH cli

llerafel committed:
Fallback for testing on fork and checkout step on release

llerafel commented:
Thanks for the review. Fixes applied:
- Removed `|| true` to ensure the build fails if binaries are missing.
- Simplified artifact upload path to `${{ env.STR_BUILD_DIR }}` 
- Added `checkout` to the release job for proper gh cli contex.

#### GHA test run
- [New tag](https://github.com/llerafel/TiltedEvolution/actions/runs/20937055862)
- [Manual/Cron](https://github.com/llerafel/TiltedEvolution/actions/runs/20936247429)

llerafel committed:
Clean up fallback on fork x2

miredirex committed:
Merge branch 'dev' into dev

miredirex committed:
No crashpad
